crates/client/db/src/rocksdb_snapshot.rs

```diff
@@ -0,0 +1,169 @@
+use librocksdb_sys as ffi;
+use rocksdb::{AsColumnFamilyRef, DBAccess, DBPinnableSlice, Error, ReadOptions};
+use std::sync::Arc;
+
+/// A copy of [`rocksdb::SnapshotWithThreadMode`] with an Arc<DB> instead of an &'_ DB reference
+pub struct SnapshotWithDBArc<D: DBAccess> {
```

Comment:
Can you provide more information for why this needs to be done. Your comment in the PR is good, but it also needs to be in the code. Links to the GH issues you referenced would be helpful as well.